crates/oxc_linter/src/context.rs

```diff
@@ -34,8 +73,68 @@ pub struct LintContext<'a> {
     severity: Severity,
 }
 
+#[derive(Clone)]
```

Comment:
I *think* the transmute between `&LintContext`/`&CFGLintContext` and `&LintContextData` is sound as you've made both structs `#[repr(transparent)]`.

But if someone changed the type defs without understanding this need, it'd lead to UB.

To make this unlikely to happen, I think it needs a comment here at the location of the type definitions explaining what is going on here, the motivation for it, and saying at minimum "It is **essential** that `LintContext` and `LintContextData` have identical memory layouts so that `From` impls below are sound.".